app/src/main/java/com/jwisozk/flightreviews/SubmitFragment.kt

```diff
@@ -0,0 +1,122 @@
+package com.jwisozk.flightreviews
+
+import android.os.Bundle
+import android.view.View
+import android.widget.ProgressBar
+import android.widget.RatingBar
+import androidx.constraintlayout.motion.widget.MotionLayout
+import androidx.coordinatorlayout.widget.CoordinatorLayout
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.appbar.AppBarLayout
+import com.google.android.material.snackbar.Snackbar
+
+
+class SubmitFragment : Fragment(R.layout.fragment_submit) {
+
+    companion object {
+        @JvmStatic
+        fun newInstance() = SubmitFragment()
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val activity = requireActivity() as MainActivity
+        val rootLayout: CoordinatorLayout = view.findViewById(R.id.submitFragment)
+        val spinner: ProgressBar = view.findViewById(R.id.spinner)
+        val repository = Repository(getNetworkService())
+        val viewModel = ViewModelProvider(activity).get(ParamViewModel::class.java)
+        viewModel.reset(ParamViewModel.Companion.Input(repository))
+        val viewManager = LinearLayoutManager(activity)
+        val ratingBar = view.findViewById<RatingBar>(R.id.appbarRatingBar)
+        ratingBar.onRatingBarChangeListener =
+            RatingBar.OnRatingBarChangeListener { _, rating, _ ->
+                viewModel.onOverallRatingChanged(rating, AbsParamCell.LabelType.FLIGHT)
+            }
+        val motionLayout = view.findViewById<MotionLayout>(R.id.motionLayout)
+        val appBar = view.findViewById<AppBarLayout>(R.id.appBarLayout)
+        appBar.addOnOffsetChangedListener(object : MotionLayout(activity),
```

Comment:
Wow, we don't need this `MotionLayout(activity)` thing. It just creates an unnecessary `View` that isn't attached to anything. Instead we should do it like this:
```kotlin
        val motionLayout = view.findViewById<MotionLayout>(R.id.motionLayout)
        val appBar = view.findViewById<AppBarLayout>(R.id.appBarLayout)
        appBar.addOnOffsetChangedListener(
            AppBarLayout.OnOffsetChangedListener { _, verticalOffset ->
                motionLayout.progress = -verticalOffset / appBar.totalScrollRange.toFloat()
            }
        )
```